examples-api-use/image-example.cc

```diff
@@ -82,14 +82,18 @@ static ImageVector LoadImageAndScaleImage(const char *filename,
 void CopyImageToCanvas(const Magick::Image &image, Canvas *canvas) {
   const int offset_x = 0, offset_y = 0;  // If you want to move the image.
   // Copy all the pixels to the canvas.
+  // Get direct access to pixel data
+  const Magick::PixelPacket *pixels = image.getConstPixels(0, 0, image.columns(), image.rows());
+
   for (size_t y = 0; y < image.rows(); ++y) {
+    const Magick::PixelPacket *row = pixels + (y * image.columns());
     for (size_t x = 0; x < image.columns(); ++x) {
-      const Magick::Color &c = image.pixelColor(x, y);
-      if (c.alphaQuantum() < 256) {
+      const auto &[blue, green, red, opacity] = row[x];
```

Comment:
This http://www.graphicsmagick.org/Magick++/PixelPacket.html

describes the PixelPacket to contain RGBA ... by you seem to deconstruct it as BGRA

In fact, it can be either https://foss.heptapod.net/graphicsmagick/graphicsmagick/-/blob/branch/default/magick/image.h?ref_type=heads#L529-548

So I don't think it is safe to do it this way, but rather access the struct directly.

Also, is the raw sample the raw row contains the same as a Quantum (as we use that assumption to scale).